packages/component_code_gen/helpers/langchain_helpers.py

```diff
@@ -106,14 +106,14 @@ def ask_agent(prompt, docs, templates, auth_example, parsed_common_files, urls_c
     return result
 
 
-def no_docs(prompt, templates, auth_example, parsed_common_files, urls_content):
+def no_docs(prompt, templates, auth_example, parsed_common_files, urls_content, normal_order=True):
     user_prompt = create_user_prompt(prompt, urls_content)
     system_instructions = format_template(
         templates.system_instructions(auth_example, parsed_common_files))
 
     result = get_llm()(messages=[
-        SystemMessage(content=system_instructions),
-        HumanMessage(content=user_prompt),
+        SystemMessage(content=system_instructions if normal_order else user_prompt),
```

Comment:
I think you should do a refactor of the `system_instructions` to be very small — just the core instructions about the bot's purpose / expected output.

Then I'd put most of the content in the `user_prompt`.

That should let you remove this logic of passing different content in the system vs. human message here. Instead, the `user_prompt` should be constructed with the standard / reverse order in `generate_component_code`:

- Standard order: component rules first, docs last
- Reverse order: docs first, component rules last

I'd like to see how that works.

Reply:
Basically I've heard that you should always put the most important content in the human message vs. the system message, so I think even that refactor may also improve the script.